app/models/grda_warehouse/tasks/service_history/enrollment.rb

```diff
@@ -60,7 +60,12 @@ def self.ensure_there_are_no_extra_enrollments_in_service_history(client_ids)
     end
 
     def invalidate_source_data!
```

Comment:
@dtgreiner @eanders I'm trying to see if households are invalidated when an enrollment is invalidate. Elliot pointed me to this change, but afaict this method `invalidate_source_data!` is not called anywhere in the codebase. Am I missing something?

For HMIS Enrollments, and it appears for WH Enrollments too, [these invalidate_processing! methods are used](https://github.com/greenriver/hmis-warehouse/blob/3b88c3a659a940fdf4cd7c2916792d36b7a0a43c/app/models/concerns/hud_concerns/enrollment.rb#L45-L55). I think those methods need to be updated to invalidate all household members.

Reply:
@gigxz `invalidate_processing!` is called in `ClientCleanup` and `ProjectCleanup`.  These are run nightly and in situations where changes are made to either the projects or clients that require rebuilds, they are used to tell `app/models/grda_warehouse/tasks/service_history/enrollment.rb` that the enrollments need to be rebuilt.  I think you are correct about `invalidate_source_data!`.  That was added to fix this issue, but I'm unable to figure out where it is called. @dtgreiner was that method maybe supposed to replace `invalidate_processing!`?

Reply:
@eanders If I'm recalling correctly, we identified the `invalidate_source_data!` method as the one supposed to be run when the service history source data changes, but it does look like `invalidate_processing!` is doing this. The `invalidate_source_data!` method appears to have been around since 2020, but looking into it more it doesn't seem to ever have been referenced. It may be worth a call if you have time to dig into this a little more.